lib/components/app/app-menu.js

```diff
@@ -41,9 +47,92 @@ class AppMenu extends Component {
     window.location.href = startOverUrl
   }
 
+  _expandedPane = () => {
+    const { isPaneOpenLeft } = this.state
+    this.setState({ isPaneOpenLeft: !isPaneOpenLeft })
+  }
+
+  _expandSubmenu = () => {
+    const { isSubmenuExpanded } = this.state
+    this.setState({ isSubmenuExpanded: !isSubmenuExpanded })
+  }
+
+  _addExtraMenuItems = (menuItems) => {
+    const AddIconAndLabel = ({iconType, iconUrl, label}) => {
+      return (
+        iconUrl
+          ? <span>
+            <img alt={`icon for ${label} Menu Item`} src={iconUrl || 'bus'} />
+            {label}
+          </span>
+          : <span><Icon name={iconType || 'car'} />{label}</span>
+      )
+    }
+
+    return (
+      menuItems.map(menuItem => {
+        const {
+          children,
+          containsChildren,
+          href,
+          iconType,
+          iconUrl,
+          id,
+          label,
+          separator
+        } = menuItem
+        return (
+          containsChildren
```

Comment:
I think an if statement/default return would be cleaner than a ternary operator

Reply:
Yes perhaps.  I'm ok with the ternary as there are many similar patterns throughout OTP RR. @binh-dam-ibigroup happy for you to have the final say?